pytorch_forecasting/metrics/base/_base_object.py

```diff
@@ -0,0 +1,105 @@
+"""Base object class for pytorch-forecasting metrics."""
+
+from pytorch_forecasting.base._base_object import _BaseObject
+
+
+class _BasePtMetric(_BaseObject):
+    """Base class for metric object that can be discovered for testing."""
+
+    _tags = {"object_type": "metric"}
+
+    @classmethod
+    def name(cls):
+        """Get the name of the metric.
+
+        Returns
+        -------
+        str
+            The name of the metric.
+        """
+        metric_cls = cls.get_metric_cls()
+        return metric_cls.__name__
+
+    @classmethod
+    def get_metric_cls(cls):
+        """Get the metric class.
+
+        Returns
+        -------
+        type
+            The metric class.
+        """
+        raise NotImplementedError("get_metric_cls must be implemented in subclasses.")
+
+    @classmethod
+    def prepare_test_inputs(cls, test_case):
+        """Prepare test inputs for the metric.
+
+        This can be overriden by subclasses to provide special handling
+        of test inputs.
+
+        Parameters
+        ----------
+        test_case: dict
+            Dictionary containing test case parameters.
+
+        Returns
+        -------
+        (y_pred, y_actual, kwargs): tuple
+            Tuple containing the predicted values, actual values, and any additional
+            keyword arguments.
+        """
+
+        return test_case["y_pred"], test_case["y"], {}
+
+    @classmethod
+    def get_test_params(cls):
+        """Returns parameters for initializing the metric for testing.
+
+        Returns
+        -------
+        dict
+            Dictionary containing parameters for initializing the metric.d
+        """
+
+        return {}
+
+    def get_metric_type(cls):
```

Comment:
this method looks superfluous, since the idiomatic call to `get_tag` is not that much longer.

Plus, `get_tag` can be called only on instances, not classes. If you want to call on classes, use `get_class_tag`.